### DISCUSSION TITLE
How to get keys and/or connection strings from a ServiceBus module?

### DISCUSSION THREAD

**Discussion by bert2** [Azure/ResourceModules#1117] [Q&A]
I'm struggling with the following scenario: In my bicep I'm creating a resource that my web app should connect to via a connection string. For instance I'm creating a [`Microsoft.ServiceBus/namespaces` module](https://github.com/Azure/ResourceModules/tree/main/arm/Microsoft.ServiceBus/namespaces). Later when I create my `Microsoft.Web/sites` web app I want to inject the connection string as an environment variable during deployment. But how can I get the connection string from a module?

```bicep
module myServiceBus './Microsoft.ServiceBus/namespaces/deploy.bicep' = {
    // ...
}

module myWebApp './Microsoft.Web/sites/deploy.bicep' = {
    // ...
    params: {
        // ...
        siteConfig: {
            // ...
            appSettings: [
                {
                    'name': 'ServiceBus__ConnectionString'
                    'value': 'how to get this value from myServiceBus?'
                }
            ]
        }
    }
}
```

Am I supposed to use [`listKeys()`](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-resource#list) on [Microsoft.ServiceBus/namespaces/authorizationRules](https://docs.microsoft.com/en-us/rest/api/servicebus/stable/namespaces-authorization-rules/list-keys)? Not sure how to go about this exactly though.

I'm getting the feeling that what I'm trying to do is discouraged (maybe for security reasons) and I'm supposed to do something else entirely.

**Comment by AlexanderSehr** [comment]
Hey @bert2, while I'm no expert for the service (and can hence not speak for the validity of the approach) I am fairly certain the current version of the `sites` module will require some update from your end to make this work. It's currently more an example than anything else.

In general, to pass a connection string from one module to another, you may be able to built it, but you could also add the connection string as an output to the service bus module and consume it in a subsequent deployment. However, this really only makes sense if the output is not considered sensitive as there is no `@secure` flag for outputs in bicep today
